### DISCUSSION TITLE
Bind Number to FluentTextField

### DISCUSSION THREAD

**Discussion by AlexHedley** [microsoft/fluentui-blazor#3078] [Q&A]
Is it possible to bind a number to a `FluentTextField`?

```html
<FluentTextField TextFieldType="TextFieldType.Number" @bind-Value=Message.MessageId Disabled="true">
    <FluentLabel Slot="start" Typo="Typography.Body">Message Id</FluentLabel>
</FluentTextField>
```

> The left-hand side of an assignment must be a variable, property or indexer

```html
<FluentTextField TextFieldType="TextFieldType.Number" @bind-Value=Message.MessageId.ToString() Disabled="true">
    <FluentLabel Slot="start" Typo="Typography.Body">Message Id</FluentLabel>
</FluentTextField>
```

- https://www.fluentui-blazor.net/TextField

---

I don't need the ^v arrows so no need for a `NumberField`.

```html
<FluentNumberField @bind-Value="exampleInt" Label="Integer" />

@code {
    int exampleInt { get; set; } = 123;
}
```

- https://www.fluentui-blazor.net/NumberField

**Comment by oneolddev** [comment]
@AlexHedley 

What are you trying to achieve? `Disabled` is set to `true` so I assume you are not allowing the user to change the value.  

Or do you simply don't want the spinners(boxed in red) not to appear when `Disabled` is set to `false`.
![image](https://github.com/user-attachments/assets/07dbcb98-83e9-4178-9a60-57ce25d6df59)

**Reply by AlexHedley** [reply]
I’m likely to have a toggle at some point where I can switch into Edit mode so thought it would be easier to have the same view and just toggle the disabled property than have `TextField` for _Edit_ and `Label` for not.

Plus sometimes the look and feel of an `input` is nicer and would be consistent with the rest of the `Form` but that can also be achieved with some overridden styling.

I guess it doesn’t really matter if the spinners are there so I should just use a `NumberField`.

I might just be used to binding any value to an `input` and it’s forgiving (bad JS libs :p). It makes more sense here to force the bound value to be specific to what you need, i.e use `DatePicker` for a _Date_ etc.

Thanks for the input :).

**Reply by vnbaaij** [reply]
There is also the `HideStep` parameter for the NumberField which...surprise, surprise hides the spin buttons

**Reply by AlexHedley** [reply]
My inability to read docs has failed me again! Thanks for letting me know about that :-). 

Do you accept PRs for adding others, if they were missing?

**Reply by vnbaaij** [reply]
Yes, please!
